src/ApiCore/Transport/RestTransport.php

```diff
@@ -156,15 +157,52 @@ private function getCallOptions(array $options)
         return $callOptions;
     }
 
+    /**
+     * @param \Exception $ex
+     * @return ApiException
+     */
     private function convertToApiException(\Exception $ex)
     {
-        $res = (string) $ex->getResponse()->getBody();
-        $rObj = (object) json_decode($res, true)['error'];
-        // Overwrite the HTTP Status Code with the RPC code, derived from the "status" field.
-        $rObj->code = ApiStatus::rpcCodeFromStatus($rObj->status);
-        $rObj->metadata = property_exists($rObj, 'details') ? $rObj->details : null;
-        $rObj->details = $rObj->message;
-
-        return ApiException::createFromStdClass($rObj);
+        $res = $ex->getResponse();
+        $body = (string) $res->getBody();
+        if ($error = json_decode($body, true)['error']) {
+            // Overwrite the HTTP Status Code with the RPC code if it exists.
+            $basicMessage = $error['message'];
+            $status = $error['status'];
+            $metadata = isset($error['details']) ? $error['details'] : null;
+        } else {
+            // Only map HTTP status codes from Google\Rpc\Code which do not map
+            // to multiple gRPC codes (e.g. excluding 400, 409, and 500).
+            $httpToRpcCodes = [
```

Comment:
What do you think about moving this conversion into `ApiStatus`? Or somewhere else that it could be reused if needed?

Reply:
sounds good, I moved this into `ApiStatus`